src/features/post/postsSlice.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getUserPosts } from '../../api/posts';
+import { Post } from '../../types/Post';
+
+type InitialState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: InitialState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadPosts = createAsyncThunk('posts/fetch', (id: number) => {
+  return getUserPosts(id);
+});
+
+export const postsSlice = createSlice({
+  name: 'post',
```

Comment:
The name property should be 'posts' instead of 'post' to be consistent with the naming of the slice and feature